public/components/common/seach/queryBar.tsx

```diff
@@ -9,28 +9,83 @@
  * GitHub history for details.
  */
 
-import React from 'react';
+import React, {Fragment, useEffect, useState, useCallback} from 'react';
 import {
+  EuiButtonEmpty,
   EuiFieldSearch,
   EuiFlexItem,
+  EuiSuggest,
+  EuiFlexGroup,
+  EuiSelectableTemplateSitewide,
+  EuiSelectableTemplateSitewideOption,
+  EuiSuggestItem,
+  EuiSuggestItemProps
 } from '@elastic/eui';
+import useDebounce from 'react-use/lib/useDebounce'
 import {
   IQueryBarProps
 } from './search';
 import {
   RAW_QUERY
 } from '../../../common/constants/explorer';
+import { _termValuesToQuery } from '@elastic/eui/src/components/search_bar/query/ast_to_es_query_dsl';
+//import {getQuerySuggestions} from './autocomplete_service'
 
-export function QueryBar(props: IQueryBarProps) {
 
+const commands: EuiSuggestItemProps[] =[
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'dedup'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'eval'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'fields'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'rename'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'search'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'sort'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'stats'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'where'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'head'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'top'},
+  {type: {iconType: 'search', color: 'tint1'}, label: 'source'}
+]
+
+export function QueryBar(props: IQueryBarProps) {
+  
   const {
     query,
```

Comment:
just curious, looks like the text field depends on `inputValue` to display, calls `handleQuerySearch` to search. what are `query`, `handleQueryChange`, `value` used for?

Reply:
the function that renders the search bar is `QueryBar` that takes in the `IQueryBarProps` props, which is declared in [search.tsx](https://github.com/sejli/trace-analytics/blob/feature/autocomplete/public/components/common/seach/search.tsx) with an interface creating `query`, `handleQueryChange`, and `handleQuerySearch`. `query` is the actual query that is sent to backend, which is updated by `handleQueryChange` each time the input changes (see `onInputChange` function). as mentioned earlier, `value` is redundant and i will be using `inputValue` to keep track of the string that the user types in. since `query` isn't a string that can be split into a list, i'm using `inputValue` to filter the suggestions.

Reply:
got it. still a bit confused because if you keep `query` up to date using `handleQueryChange` each time the input changes, then why not use `query` instead of `inputValue`?

Reply:
i split the user input using `.split(' ')`, but for some reason, it still doesn't work with `query` even though i made sure that query is a string. couldn't get it to work so i just made another `inputValue` state